packages/docs/api/equality-algorithm.md

```diff
@@ -0,0 +1,225 @@
+---
+id: equality-algorithm
+title: Equality Algorithm
+---
+
+The equality algorithm is an extensible and parameterized deep equality checker.
+It is used by all functions that check deep equality inside earl.
+
+## Algorithm
+
+To determine is two values are equal the following checks are executed:
+
+1. If the second value is a [Matcher](/guides/using-matchers.md) and it...
+   1. matches the first value, then the values are considered equal;
+   1. doesn't match the first value, then the values are considered unequal.
+1. Every [custom rules](/advanced/plugin-development.md) is checked.
+   1. If it returns undefined the next rule is checked or the algorithm
+      advances.
+   1. If it returns success, then the values are considered equal.
+   1. If it returns error, then the values are considered unequal.
+1. If any of the values is equal to one of it's ancestors (is a self-referencing
+   object)...
+   1. If the other value is equal to it's ancestor of the same distance (e.g.
+      grandparent and grandparent), then the values are considered equal.
+   2. Otherwise they are considered unequal.
+1. A category is determined for each value based on the rules of each category.
+   1. If the values have different categories, then they are considered unequal.
+   1. If the category is the same the values are compared according to rules of
+      that category.
+
+Additionally the algorithm takes the following options:
+
+- `uniqueNaNs` - Considers two `NaN` values unequal, defaults to `false`.
```

Comment:
Are the options ever exposed for the user?

Reply:
Currently they aren't. I think in the future earl could change the API surface from `isEqual` and `isLooseEqual` to `isEqual(value, options?)` which would expose them. I kept `uniqueNaNs`,`minusZero` and `compareErrorStack` even though they are set to `false` everywhere, because they are design choices that I felt are best exposed rather then hidden.